addons/stock/models/stock_move.py

```diff
@@ -250,11 +250,11 @@ def _compute_has_move_lines(self):
         for move in self:
             move.has_move_lines = bool(move.move_line_ids)
 
-    @api.one
     @api.depends('product_id', 'product_uom', 'product_uom_qty')
     def _compute_product_qty(self):
-        rounding_method = self._context.get('rounding_method', 'UP')
-        self.product_qty = self.product_uom._compute_quantity(self.product_uom_qty, self.product_id.uom_id, rounding_method=rounding_method)
+        for move in self:
+            rounding_method = move._context.get('rounding_method', 'UP')
+            move.product_qty = move.product_uom._compute_quantity(move.product_uom_qty, move.product_id.uom_id, rounding_method=rounding_method)
```

Comment:
The definition of `rounding_method` only depends on the context, and can thus be moved outside the loop.